### ISSUE TITLE
"Invoking "make -j8 -l8" failed" when doing "catkin_make"

### ISSUE THREAD

**Issue by T1mLen6** [leap-hand/LEAP_Hand_API#4]
Hi,

I got this error when I was trying to run the catkin_make command:

I have also ran the rosdep check and seems all dependencies have been satisfied:
```
(leap) (base) tim@tim-mail-32:~/catkin_ws$ rosdep check --from-paths /home/tim/catkin_ws/src --ignore-src
All system dependencies have been satisfied
```

Below is the original error message:
(leap) (base) tim@tim-mail-32:~/catkin_ws$ catkin_make
Base path: /home/tim/catkin_ws
Source space: /home/tim/catkin_ws/src
Build space: /home/tim/catkin_ws/build
Devel space: /home/tim/catkin_ws/devel
Install space: /home/tim/catkin_ws/install
####
####
####
####
[  0%] Built target std_msgs_generate_messages_eus
[  0%] Built target std_msgs_generate_messages_py
[  0%] Built target std_msgs_generate_messages_lisp
[  0%] Built target std_msgs_generate_messages_nodejs
[  0%] Built target std_msgs_generate_messages_cpp
[  0%] Built target _leap_hand_generate_messages_check_deps_leap_effort
[  0%] Built target _leap_hand_generate_messages_check_deps_leap_position
[  0%] Built target _leap_hand_generate_messages_check_deps_leap_velocity
[ 17%] Built target leap_hand_generate_messages_nodejs
[ 41%] Built target leap_hand_generate_messages_py
[ 82%] Built target leap_hand_generate_messages_lisp
[ 82%] Built target leap_hand_generate_messages_eus
[ 88%] Generating C++ code from leap_hand/leap_velocity.srv
[ 94%] Generating C++ code from leap_hand/leap_position.srv
[100%] Generating C++ code from leap_hand/leap_effort.srv
Traceback (most recent call last):
Traceback (most recent call last):
  File "/opt/ros/noetic/share/gencpp/cmake/../../../lib/gencpp/gen_cpp.py", line 49, in <module>
  File "/opt/ros/noetic/share/gencpp/cmake/../../../lib/gencpp/gen_cpp.py", line 49, in <module>
Traceback (most recent call last):
  File "/opt/ros/noetic/share/gencpp/cmake/../../../lib/gencpp/gen_cpp.py", line 49, in <module>
    genmsg.template_tools.generate_from_command_line_options(
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 213, in generate_from_command_line_options
    genmsg.template_tools.generate_from_command_line_options(
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 213, in generate_from_command_line_options
    genmsg.template_tools.generate_from_command_line_options(
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 213, in generate_from_command_line_options
    generate_from_file(argv[1], options.package, options.outdir, options.emdir, options.includepath, msg_template_dict, srv_template_dict)
    generate_from_file(argv[1], options.package, options.outdir, options.emdir, options.includepath, msg_template_dict, srv_template_dict)
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 156, in generate_from_file
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 156, in generate_from_file
    generate_from_file(argv[1], options.package, options.outdir, options.emdir, options.includepath, msg_template_dict, srv_template_dict)
    _generate_srv_from_file(input_file, output_dir, template_dir, search_path, package_name, srv_template_dict, msg_template_dict)
    _generate_srv_from_file(input_file, output_dir, template_dir, search_path, package_name, srv_template_dict, msg_template_dict)
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 156, in generate_from_file
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 109, in _generate_srv_from_file
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 109, in _generate_srv_from_file
    _generate_from_spec(input_file,
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 77, in _generate_from_spec
    _generate_from_spec(input_file,
    interpreter = em.Interpreter(output=ofile, globals=g, options={em.RAW_OPT:True,em.BUFFERED_OPT:True})
    _generate_srv_from_file(input_file, output_dir, template_dir, search_path, package_name, srv_template_dict, msg_template_dict)
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 77, in _generate_from_spec
                                                                   ^^^^^^^^^^
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 109, in _generate_srv_from_file
AttributeError: module 'em' has no attribute 'RAW_OPT'
    interpreter = em.Interpreter(output=ofile, globals=g, options={em.RAW_OPT:True,em.BUFFERED_OPT:True})
                                                                   ^^^^^^^^^^
    _generate_from_spec(input_file,
  File "/opt/ros/noetic/lib/python3/dist-packages/genmsg/template_tools.py", line 77, in _generate_from_spec
AttributeError: module 'em' has no attribute 'RAW_OPT'
    interpreter = em.Interpreter(output=ofile, globals=g, options={em.RAW_OPT:True,em.BUFFERED_OPT:True})
                                                                   ^^^^^^^^^^
AttributeError: module 'em' has no attribute 'RAW_OPT'
make[2]: *** [ros_module/CMakeFiles/leap_hand_generate_messages_cpp.dir/build.make:66: /home/tim/catkin_ws/devel/include/leap_hand/leap_velocity.h] Error 1
make[2]: *** Deleting file '/home/tim/catkin_ws/devel/include/leap_hand/leap_velocity.h'
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [ros_module/CMakeFiles/leap_hand_generate_messages_cpp.dir/build.make:80: /home/tim/catkin_ws/devel/include/leap_hand/leap_effort.h] Error 1
make[2]: *** Deleting file '/home/tim/catkin_ws/devel/include/leap_hand/leap_effort.h'
make[2]: *** [ros_module/CMakeFiles/leap_hand_generate_messages_cpp.dir/build.make:73: /home/tim/catkin_ws/devel/include/leap_hand/leap_position.h] Error 1
make[2]: *** Deleting file '/home/tim/catkin_ws/devel/include/leap_hand/leap_position.h'
make[1]: *** [CMakeFiles/Makefile2:788: ros_module/CMakeFiles/leap_hand_generate_messages_cpp.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
Invoking "make -j8 -l8" failed

Many thanks!

**Comment by Lucky7Chess** [comment]
This isn't really the scope of LEAP Hand since it is a ROS issue.  However, your error is AttributeError: module 'em' has no attribute 'RAW_OPT'.  Did you install empy properly?  Maybe you have em installed but you shouldn't?

**Comment by T1mLen6** [comment]
Ok never mind, issue solved. Turns out empy just released their 4.0 month ago and somehow they deleted the RAW_OPT Interpreter option, using pip install empy==3.3.4 solved that issue, thanks!

**Comment by Lucky7Chess** [comment]
Got it I'll update the readme, thanks!
